hazelcast/src/test/java/com/hazelcast/wan/impl/WanPublisherMigrationTest.java

```diff
@@ -125,8 +124,12 @@ public void testMigration() {
             assertTrue("Expected at least " + partitionsToMigrate
                             + " migration operations to be processed but was " + publisher,
                     publisher.migrationProcess.intValue() >= partitionsToMigrate);
-            assertEquals(exceptionMsg("migrationCommit", publisher),
-                    partitionsToMigrate, publisher.migrationCommit.intValue());
+
+            //onMigrationCommit() method of the listener is called asynchronously. So assert eventually
+            assertTrueEventually(() -> assertEquals(exceptionMsg("migrationCommit", publisher),
```

Comment:
Eventual assertion fixes this test but this seems odd to me, because there should be no need for eventual assertion after graceful shutdown. I went a little back in versions, by starting with 4.1, the expected behavior seems changed.
Better not to fix this test with eventual assertion till we understand what's going on. 

Waiting @vbekiaris view on this potential root issue.